docs/dev_cycle.part-01.rst

```diff
@@ -0,0 +1,288 @@
+
+**Visibility:**
+The ``public`` decleration ensures that all state variables and the ``donate`` function will be available for the callers
+of the contrat, in the contract's interface.
+
+.. note::   
+    For the public state variables, the compiler actually creates an accessor function
+    which if you had to type manually could look like: ``function total() public returns (uint) {return donations_total;}``
+
+**Data types**:
+Since we are dealing with numbers, the only data type we use here is ``uint``, unsigned integer. The ``int`` and ``uint``
+are declated in steps of 8 bits, ``unint8``, ``uint16`` etc. When the bits indicator is omitted, like ``int`` or ``uint``, the compiler will
+assumes ``uint256``.
+
+.. note:: 
+
+    If you know in advance the the maximum size of a variable, better to limit the type and save the gas of extra 
+    memory or storage.
+    
+As of version 0.4.17 Solidity does *not* support decimal point types. If you need decimal point, you will have to manauly handle
+the fixed point calculations with integers. For the sake of simplicty, the example uses only ints.
+
+    
+**Constructor**:
+The function ``function Donator()`` is a constructor. A constructor function's name is always identical to the contract's name. 
+It runs once, when the contract is created, and can't be called again. Here we set the ``default_usd_rate``, to be used
+when the donator didn't provide the effective exchange rate. Providing a constructor function is optional.
+
+
+**Functions**:
+The ``donate`` function accepts one argument: ``usd_rate``. Then the function
+updates the total donated, both of Ether and USD value. It also updates the default USD rate and the donations counter.
+
+**Magic Variables**:
+In every contract you get three magic variables in the global scope: ``msg``, ``block`` and ``tx``. You can use these
+variable without prior decleration or assignment. To find out how much 
+Ether was sent, use ``msg.value``.
+
+**Modifiers**:
+``modifier money_sent() { if (!msg.value > 0) throw; _; }``. The term "modifier" is a bit confusing. 
+A modifier of a function is  *another* function that injects, or modifies, code, typically to verify some pre-existing condition.
+Since the donate function uses the modifier ``function donate(uint usd_rate) public payable money_sent {...}``,
+then ``money_sent`` will run *before* ``donate``. The code in ``donate`` will run only if ``msg.value > 0``, and make sure
+that the ``donations_count`` does not increase by a zero donation.
+
+.. note::
+    
+    The modifier syntax uses ``_;`` to tell solidity where to insert the *modified* function.
+    We can of course check the include the modifier condition the original function, but a declared modifier is handy
+    when you want to use the same pre-condition validation in more than one function.
+
+**Fallback**:
+The weired function without a name, ``function () payable {...}``, is the "fallback". It calls ``donate``, so when somebody just
```

Comment:
suggest removing the word *weirded*(both not a word and I think it takes away from legability)